### PR TITLE
Enable doctests using xdoctest conventions

### LINKED ISSUE #506: Most doctests are not written to be tested
I'm currently integrating some better doctest handling into the tox builds and have run up against a glaring issue: Many of the documentation examples are reliant on fictional files, for instance in `subset.py`:
```
    Examples
    --------
    >>> import geopandas as gpd
    >>> ds = xarray.open_dataset(path_to_tasmin_file)  # This file doesn't exist --> Failure
    >>> polys = gpd.read_file(path_to_multi_shape_file) # This file doesn't exist --> Failure

    Get a mask from all polygons in the shape file
    >>> mask = create_mask_vectorize(x_dim=ds.lon, y_dim=ds.lat, poly=polys)
    >>> ds = ds.assign_coords(regions=mask)
```
and for the documented examples in `ensembles.py`, they effectively all must be skipped:
```
    Examples
    --------
    Create ensemble dataset
    >>> ens = create_ensemble(temperature_datasets)  # doctest: +SKIP  # This file doesn't exist --> Failure

    Calculate default ensemble percentiles
    >>> ens_percs = ensemble_percentiles(ens)  # doctest: +SKIP # The previous line was skipped so all subsequent lines must be as well
    ...
```
I can skip all the tests that can't be tested for the moment, but unless we add some very small bunk data to the main package or run doctests only when the source files are present (which isn't so much issue for Travis CI or when building from the release source files in `.tar.gz`), most of these tests will need to be skipped.

Wondering if we have any ways of envisioning how we structure doctests. I will link to the PR here once it's been opened, so we can see exactly how many tests are passing/failing.

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #506 
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] `bumpversion (minor / major / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 

This PR re-enables the doctests that were previously skipped with #507. Minor updates to API calls have been made in order for examples to be synonymous with functions (_the reason for doctests_, one could say). With these changes, all `Examples` in the library are called and evaluated. 

* **Does this PR introduce a breaking change?** 

No. `xdoctests` was already a required testing dependency and functional codes have not been touched.

* **Other information**:

Two tests are presently being troublesome:
- `tests/indices/_multivariate.py::cold_spell_duration_index`, as written, is hanging (this could be due to the need for more bunk tasmin data).
- `tests/ensembles.py::kmeans_reduce_ensemble` is giving some esoteric TypeError traceback that I can't make sense of. 
- `tests/subset.py` are all being skipped, as this module is being deprecated very soon.

### CODE REVIEW

**Comment by Zeitsperre** [comment]
Alright, I'll set the failing ones to `+SKIP` for the time being in order for the builds to pass properly (we'll still know which ones are being skipped). Will perform the rest of the PR merge checks now.

File: xclim/ensembles.py [hunk 1]
```diff
@@ -718,11 +718,11 @@ def plot_rsqprofile(fig_data):
...
-    >>> from xclim.ensembles import kmeans_reduce_ensemble, plot_rsqprofile  # doctest: +SKIP
-    >>> crit = xr.open_dataset(path_to_ensemble_file).data  # doctest: +SKIP
-    >>> ids, cluster, fig_data = kmeans_reduce_ensemble(data=crit, method={'rsq_cutoff':0.9}, random_state=42)  # doctest: +SKIP
-    >>> plot_rsqprofile(fig_data)  # doctest: +SKIP
+    >>> from xclim.ensembles import kmeans_reduce_ensemble, plot_rsqprofile
+    >>> is_matplotlib_installed()
+    >>> crit = xr.open_dataset(path_to_ensemble_file).data
+    >>> ids, cluster, fig_data = kmeans_reduce_ensemble(data=crit, method={'rsq_cutoff':0.9}, random_state=42)
+    >>> plot_rsqprofile(fig_data)
     """
```
Comment (Zeitsperre) [inline xclim/ensembles.py:726]:
```suggestion
    """
        if not MPL_INSTALLED:
        logging.error("Matplotlib is not installed. Unable to execute plotting functions.")
        raise ImportError("matplotlib")

```
Is this a terrible idea?

Comment (aulemahal) [reply xclim/ensembles.py:726]: bah... Not a bad idea, but isn't the name "plt_rsqprofile" already quite explicit on the fact that it does plot? (And thus needs matplotlib)?

Comment (Zeitsperre) [reply xclim/ensembles.py:726]: I suppose so. Nothing prevents this from being called as is. If MPL isn't there, I suppose the errors from python ("plt is not defined" or something like that) would be enough?

Comment (huard) [reply xclim/ensembles.py:726]: Yes, seems overkill. Also, maybe we should avoid creating graphics in the doctests. If it's important to show these graphics, I would place them in the notebook tutorials instead.

Comment (aulemahal) [reply xclim/ensembles.py:726]: True! We could have a notebook on all "ensembles" things, including kmeans and kkz selection.

Comment (Zeitsperre) [reply xclim/ensembles.py:726]: See #522

**Pull request merged**